### PR TITLE
Move caching into own module

### PR DESCRIPTION
**Pull request by cbmono**
### Changelog
- Move HTTP 1.1 Cache into its own singleton module (keeps the original `rq` implementation cleaner)
- Following the same principle, moved the HTTP Cache Tests into their own spec file
- Removed unused dependencies
- Added `moment` as a dependency
### TODO's
- Tests are too slow
- README.md has to be updated
- Test coverage (Istanbul) went down to ~95% (was at 100%)

### CODE REVIEW

File: lib/rp.js [hunk 1]
```diff
@@ -71,15 +59,16 @@ function RP$callback(err, response, body) {
 
     // 304 response means we can reuse the cache entry we already had
     if (response && response.statusCode === 304 && self._rp_options.cache) {
```
Comment (cbmono) [inline lib/rp.js]: This if-block should be part of `httpCache`

File: lib/rp.js [hunk 2]
```diff
@@ -245,37 +135,37 @@ request.Request.prototype.init = function RP$initInterceptor(options) {
...
 
                     self._rp_promise = new Bluebird(function (resolve) {
-                      resolve(isResolveWithFullResponseSet ? cachedResponse.content : cachedResponse.content.body);
+                      resolve(httpCache.isResolveWithFullResponseSet ? cachedResponse.content : cachedResponse.content.body);
                     });
 
                     return;
+
                 } else {
                     // Add pre-flight header
```
Comment (cbmono) [inline lib/rp.js]: This could also be moved into `httpCache `

Comment (bambata) [inline lib/rp.js]: but adding those headers should be done before the request is triggered, isn't it ?

File: lib/rp.js (hunk 1 above)
Comment (bambata) [inline lib/rp.js]: yes you r right. doing it

**Pull request closed without merging**